### ISSUE TITLE
Tekton plugin should show tekton tab and Latest PLR in CI/CD tab if annotations `janus-idp.io/tekton-enabled : 'true'` is true in entity

### ISSUE THREAD

**Issue by invincibleJai** [janus-idp/backstage-plugins#700]
**Describe the bug**
Tekton plugin show tekton tab even if the user has `janus-idp.io/tekton-enabled : 'false'` in the entity defined 

**To Reproduce**

- Setup Tekton plugin, can follow https://github.com/janus-idp/backstage-plugins/tree/main/plugins/tekton#readme 
- Add annotation `janus-idp.io/tekton-enabled : 'false'`

**Expected behavior**
Tekton tab should not be shown if annotation `janus-idp.io/tekton-enabled : 'true'` is false neither in ci/cd tab nor in tekton tab and README should get updated.

**Screenshots**

**Additional context**

**Comment by christophe-f** [comment]
Also to be consistent with all the other CI plugins, the Tekton plugin should be in the CI tab. And we can remove the Tekton tab.

**Comment by invincibleJai** [comment]
> Also to be consistent with all the other CI plugins, the Tekton plugin should be in the CI tab. And we can remove the Tekton tab.

Yeah, we can take this opportunity to improve this experience. Currently, we show only the latest PLR in CI/CD tab and in tekton tab, we show a complete list of pipeline runs with the option to drill into each so we need to define how we would like to have this experience together in CI/CD TAB

Below are some screenshots for reference

- In CI/CD tab, showing only latest pipeline run, with CTA on the footer `Go to tekton` that takes user to tetkon tab

<img width="1726" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/b3650b8d-871a-4b6c-9811-20fc2f99205f">

- A dedicated tab called `Tekton` shows a list of all pipelines run with the option to drill into these

<img width="1728" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/905c0ada-9544-4e1b-b68e-32bac6876277">

<img width="1728" alt="image" src="https://github.com/janus-idp/backstage-plugins/assets/5129024/754109c5-9d94-44a3-8282-5dbf796f00f9">

@yuvalgalanti / @sonyccd would need your help with improving this experience

cc @debsmita1 @divyanshiGupta

**Comment by invincibleJai** [comment]
> Should this be moved to the https://github.com/janus-idp/backstage-showcase repository? Or are there any plugin code changes needed?

@tumido this will need change in tekton plugin

**Comment by invincibleJai** [comment]
@christophe-f we had a discussion on this with UXD and we can take it in two-part

1. If catalog info has annotation `janus-idp.io/tekton-enabled: true` only then the Tekton tab and tekton in CI tab
2. We'll work with UXD to get the experience needed to have complete tekton in CI rather than a dedicated tab. 

LMK if you see any issues.

cc @yuvalgalanti @debsmita1

**Comment by christophe-f** [comment]
It's fine to me to split this into 2 issues. For point 2, that's something we wanted to change for a while.

For the annotation, usually it does not contains true but a value. For example a Git repo name, or k8s label, etc. It 'feels' weird to have the annotation and specify 'true'. Just having the annotation means that it is true.

**Comment by invincibleJai** [comment]
> It's fine to me to split this into 2 issues. For point 2, that's something we wanted to change for a while.
> 

Yeah we will get this addressed soon.

> For the annotation, usually it does not contains true but a value. For example a Git repo name, or k8s label, etc. It 'feels' weird to have the annotation and specify 'true'. Just having the annotation means that it is true.

Got your point. We use the k8s backend plugin to get the data. So annotations come as part of that config from k8s like https://backstage.io/docs/features/kubernetes/configuration#labeling-kubernetes-components thus we don't have any specific annotation for tekton and decided to go with `janus-idp.io/tekton-enabled` to give corntrol from entity to show/hide (This was discussed earlier here https://github.com/janus-idp/backstage-plugins/pull/242#discussion_r1173814075). We'll get back to this on how best we can handle it and Let us know if you have some suggestions.

**Comment by invincibleJai** [comment]
Have created the story here https://github.com/janus-idp/backstage-plugins/issues/804 to track new user experience as discussed. 

cc @christophe-f
